src/Stack/Clean.hs

```diff
@@ -42,15 +42,19 @@ clean (CleanOpts targets) = do
                         then Map.elems locals -- default to cleaning all local packages
                         else mapMaybe (`Map.lookup` locals) targets
             forM_ lpvs $ \(LocalPackageView{lpvRoot = pkgDir},_) -> do
-                distDir <- distDirFromDir pkgDir
-                removeTreeIfExists distDir
+                let delDir =
+                          if doFullClean
+                              then workDirFromDir pkgDir
+                              else distDirFromDir pkgDir
+                removeTreeIfExists =<< delDir
```

Comment:
Thank you point this out.
For the case if no package specified ie `stack clean --full` it would be obvious to just remove the project-wide .stack-work dir.
If package(s) specified the things became more interesting. I currently investigate the way to handle this case correctly.
If you could give a suggestions it would be much appreciated.